php/libraries/NDB_BVL_Instrument_LINST.class.inc

```diff
@@ -196,8 +196,18 @@ class NDB_BVL_Instrument_LINST extends \NDB_BVL_Instrument
                     echo "<p><b>$elname</b><br> ";
                 }
                 $rules_array = [];
-                foreach ($registered_rules as $registered) {
-                    $rules = $registered['rules'];
+                $rules       = [];
+                $registered  =[];
+                foreach ($registered_rules as $key => $value) {
+                    if ($key = 'rules') {
```

Comment:
This is changing the semantics of the function quite significantly. What are you trying to solve?

Reply:
php/libraries/NDB_BVL_Instrument_LINST.class.inc:200 PhanTypeMismatchDimFetch When fetching an array index from a value of type list<mixed>|string, found an array index of type 'rules', but expected the index to be of type int
php/libraries/NDB_BVL_Instrument_LINST.class.inc:221 PhanTypeMismatchDimFetch When fetching an array index from a value of type list<mixed>|string, found an array index of type 'group', but expected the index to be of type int
php/libraries/NDB_BVL_Instrument_LINST.class.inc:222 PhanTypeMismatchDimFetch When fetching an array index from a value of type list<mixed>|string, found an array index of type 'group', but expected the index to be of type int
php/libraries/NDB_BVL_Instrument_LINST.class.inc:225 PhanTypeMismatchDimFetch When fetching an array index from a value of type list<mixed>|string, found an array index of type 'message', but expected the index to be of type int
[debug] Restarted process exited 1

Reply:
from what I can tell phan isn't inferring the type of `$this->XINRules[$elname]` properly based on the format it's generated in in XINRegisterRule. Your fix will satify phan but break the rules themselves. 

I managed to fix phan without changing the logic here by telling it that it's an array (since it seems to always be an array in XINRegisterRule)

```
$registered_rules = $this->XINRules[$elname];
'@phan-var array $registered_rules';
```